ingestion/src/metadata/ingestion/source/databricks.py

```diff
@@ -66,6 +66,11 @@ def _get_column_rows(self, connection, table_name, schema):
 def get_columns(self, connection, table_name, schema=None, **kw):
     # This function overrides the sqlalchemy_databricks._dialect.DatabricksDialect.get_columns
     # to add support for struct, array & map datatype
+
+    # The Database name is present in the URL on the SqlAlchemy engine object. This value is the same as
+    # the one present in databricks.json in the 'database' field. This approach, while it works, may still not be optimal.
+    db_name = re.sub(r"^.+/([^/]+)$", r"\1", str(connection.engine.url))
```

Comment:
Instead of getting db name from here -> let's use the one from the config.
If the user passes it in source config `self.config.database`, we get the db name, if not we can use our usual query without DB and add error handling.

Reply:
Updated to pass database name from sqlsource via `**kw`.